app/domain/authentication/resource_restrictions/get_restrictions_from_annotation.rb

```diff
@@ -0,0 +1,48 @@
+require 'command_class'
+
+module Authentication
+  module ResourceRestrictions
+    # Parses the given annotation name and tries to extract a restriction from it.
+    # A restriction can have 2 formats:
+    # 1. "<Authenticator name>/<Restriction name>"
+    # 2. "<Authenticator name>/<Service ID>/<Restriction name>"
+    #
+    # The first format is a general restriction.
+    # This means it is relevant to all services of this authenticator.
+    #
+    # The second format is specific to the specified service ID.
+    # This means all other services will ignore it.
+    # Moreover, if both formats are found for the same restriction, the specific one will be used.
+    #
+    # This function returns the restriction name, if found (nil otherwise) and a
+    # boolean indicating if it is a general restriction or not.
+    GetRestrictionsFromAnnotation = CommandClass.new(
+      dependencies: {
+        regex_class: Regexp
```

Comment:
Do you see any value in this dependency?
Even for testing this class can be left as is.
You don't even have to have a class for regex matching, ruby knows to handle it in comparisons like `=~`.
See here for more info: https://www.rubyguides.com/2015/06/ruby-regex/

Reply:
Removed from dependencies

Reply:
I think i stay with Regexp because we extract se4rvice id and restriction name from the annotation
I see it can be done also using Struct but i prefer the one line Regexp provides
![image](https://user-images.githubusercontent.com/73157235/118607102-e0ea7e00-b7c0-11eb-9ed4-24e2d1d841bd.png)